students/Sally_Shen/lesson04/assignment/src/basic_operations.py

```diff
@@ -0,0 +1,127 @@
+import peewee
+import logging
+
+logger = logging.getLogger("basic_operations")
+logger.setLevel(logging.DEBUG)
+fh = logging.FileHandler("db.log")
+fh.setLevel(logging.DEBUG)
+logger.addHandler(fh)
+
+database = peewee.SqliteDatabase("customers.db")
+database.connect()
+database.execute_sql('PRAGMA foreign_key = ON;')
+database.execute_sql('drop table if exists customer;')
+
+
+class BaseModel(peewee.Model):
+    class Meta:
+        database = database
+
+
+class Customer(BaseModel):
+    """
+    This class defines a Customer in the databaase
+        Customer ID.
+        Name.
+        Lastname.
+        Home address.
+        Phone number.
+        Email address.
+        Status (active or inactive customer).
+        Credit limit.
+    """
+    customer_id = peewee.CharField(primary_key=True, max_length=30)
+    name = peewee.CharField(max_length=50)
+    last_name = peewee.CharField(max_length=50, null=True)
+    address = peewee.CharField(max_length=75)
+    phone_number = peewee.CharField(max_length=15)
+    email = peewee.CharField(max_length=320) # based on max email address length search
+    status = peewee.CharField(max_length=10) # "Inactive" or "Active"
+    credit_limit = peewee.FloatField()
+
+
+def create_tables():
+    with database.transaction():
+        logger.debug("Creating table Customer")
+        database.create_tables([Customer])
+
+
+def drop_tables():
+    with database.transaction():
+        logger.debug("Dropping table Customer")
+        database.drop_tables([Customer])
+
+
+def add_customer(customer_id, name, last_name, address, phone_number, email, status, credit_limit):
+    try:
+        with database.transaction():
+            customer = Customer.create(
+                customer_id=customer_id,
+                name=name,
+                last_name=last_name,
+                address=address,
+                phone_number=phone_number,
+                email=email,
+                status=status,
+                credit_limit=credit_limit
+            )
+            logger.debug(f"Customer saved {customer_id}")
+
+    except Exception as e:
```

Comment:
Be careful when you catch generic Exceptions. You typically want to only catch specific errors so you know your program is behaving as intended. Otherwise you may be capturing fatal errors that should be stopped